mappings/net/minecraft/recipe/crafting/ShapedRecipe.mapping

```diff
@@ -12,14 +12,33 @@ CLASS bal net/minecraft/recipe/crafting/ShapedRecipe
 	FIELD d output Laxx;
 	FIELD e id Lqi;
 	FIELD f group Ljava/lang/String;
+	METHOD <init> (Lqi;Ljava/lang/String;IILff;Laxx;)V
+		ARG 1 id
+		ARG 2 group
+		ARG 3 width
+		ARG 4 height
+		ARG 5 ingredients
+		ARG 6 output
 	METHOD a matchesSmall (Laug;IIZ)Z
+		ARG 1 inv
 		ARG 2 offsetX
 		ARG 3 offsetY
-	METHOD a deserializeItemStack (Lcom/google/gson/JsonObject;)Laxx;
+	METHOD a stackFromJson (Lcom/google/gson/JsonObject;)Laxx;
 		ARG 0 obj
-	METHOD b deserializePattern (Lcom/google/gson/JsonArray;)[Ljava/lang/String;
+	METHOD a findNextIngredient (Ljava/lang/String;)I
+		ARG 0 pattern
+	METHOD a makePatternList ([Ljava/lang/String;)[Ljava/lang/String;
+		ARG 0 patternLines
+	METHOD b patternFromJson (Lcom/google/gson/JsonArray;)[Ljava/lang/String;
 		ARG 0 arr
-	METHOD c deserializeComponents (Lcom/google/gson/JsonObject;)Ljava/util/Map;
+	METHOD b findNextIngredientReverse (Ljava/lang/String;)I
+		ARG 0 pattern
+	METHOD b patternToList ([Ljava/lang/String;Ljava/util/Map;II)Lff;
```

Comment:
`getIngredients`?

Reply:
This one's a bit strange, it doesn't find a `Ingredient`, it just gets the next non-empty character in the pattern string (eg. `"ss "` returns 1 in reverse or 0 forwards))